packages/sdk/src/modules/moduleFromEndpoints/utils/getHttpClient.ts

```diff
@@ -0,0 +1,60 @@
+import { Options, HTTPClientConfig } from "../types";
+
+export const getHTTPClient = (options: Options) => {
+  const getUrl = (path: string, config: HTTPClientConfig): string => {
+    const { apiUrl, ssrApiUrl } = options;
+    const { method = "POST", params = [] } = config;
+
+    // Determine the base URL based on the environment
+    const baseUrl =
+      typeof window === "undefined" ? ssrApiUrl || apiUrl : apiUrl;
+
+    // Ensure the base URL ends with a slash
+    const normalizedBaseUrl = baseUrl.endsWith("/") ? baseUrl : `${baseUrl}/`;
+    const url = `${normalizedBaseUrl}${path}`;
+    const queryParams = method === "GET" ? params : undefined;
+
+    // If there are no query params, return the URL as is
+    if (!queryParams) {
+      return url;
+    }
+
+    // If there are query params, append them to the URL as `?body=[<strignified query params>]`
+    const queryParamsString = JSON.stringify(queryParams);
+    const urlWithParams = new URL(url);
+    urlWithParams.searchParams.append("body", queryParamsString);
+
+    return urlWithParams.toString();
+  };
+
+  const getConfig = (config: HTTPClientConfig): HTTPClientConfig => {
+    const { defaultRequestConfig = {} } = options;
+    const { method = "POST", headers = {}, params = [] } = config;
```

Comment:
I can see you're setting these defaults in both `getConfig` and `getUrl`. Perhaps a single `finalConfig` object on top of the `getHTTPClient` function would be a better option?

```ts
const finalConfig = Object.assign(config, {
  method: "POST",
  params: [],
  headers: {},
});

// then

httpClient(getUrl(methodName, finalConfig), getConfig(finalConfig));
```

Reply:
Now it should be fine :)